vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/helpers/ResponseModeParametersFactory.kt

```diff
@@ -0,0 +1,85 @@
+package at.asitplus.wallet.lib.oidc.helpers
+
+import at.asitplus.KmmResult
+import at.asitplus.KmmResult.Companion.wrap
+import at.asitplus.wallet.lib.oidc.AuthenticationRequest
+import at.asitplus.wallet.lib.oidc.AuthenticationRequestParameters
+import at.asitplus.wallet.lib.oidc.OpenIdConstants
+import at.asitplus.wallet.lib.oidvci.OAuth2Exception
+import io.github.aakira.napier.Napier
+
+
+internal object ResponseModeParametersFactory {
+    fun createResponseModeParameters(
+        request: AuthenticationRequest,
+    ): KmmResult<ResponseModeParameters> = KmmResult.runCatching {
+        when (request.parameters.responseMode) {
+            null, // default for vp_token and id_token is fragment
+            OpenIdConstants.ResponseMode.FRAGMENT -> createFragmentResponseModeParameters(request.parameters)
+
+            OpenIdConstants.ResponseMode.DIRECT_POST -> createDirectPostResponseModeParameters(request.parameters)
+            OpenIdConstants.ResponseMode.DIRECT_POST_JWT -> createDirectPostJwtResponseModeParameters(
+                request,
+            )
+            OpenIdConstants.ResponseMode.QUERY -> createQueryResponseModeParameters(request.parameters)
+
+            is OpenIdConstants.ResponseMode.OTHER -> TODO()
+        }
+    }.wrap()
+
+    private fun createDirectPostResponseModeParameters(requestParameters: AuthenticationRequestParameters): ResponseModeParameters.DirectPost {
+        val responseUrl = validatePostTypeResponseModeParametersAndExtractResponseUrl(requestParameters)
+
+        return ResponseModeParameters.DirectPost(
+            responseUrl = responseUrl
+        )
+    }
+
+    private fun createDirectPostJwtResponseModeParameters(
+        request: AuthenticationRequest,
+    ): ResponseModeParameters.DirectPostJwt {
+        val responseUrl = validatePostTypeResponseModeParametersAndExtractResponseUrl(request.parameters)
```

Comment:
Please convert to expression body

Reply:
fixed in commit 9670de2a343024119e709016f74bb7cb8bcc9061